src/execution_algorithms/ltfb/truncation_selection_exchange.cpp

```diff
@@ -0,0 +1,311 @@
+#include "lbann/trainers/trainer.hpp"
+#include "lbann/utils/exception.hpp"
+#include "lbann/utils/memory.hpp"
+#include "lbann/weights/data_type_weights_impl.hpp"
+
+#include <training_algorithm.pb.h>
+
+#include <algorithm>
+#include <iterator>
+#include <memory>
+#include <sstream>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace lbann {
+namespace ltfb {
+namespace {
+
+bool low_score_wins(
+                TruncationSelectionExchange::metric_strategy strategy)
+{
+  using MetricStrategy = TruncationSelectionExchange::metric_strategy;
+  switch (strategy) {
+  case MetricStrategy::LOWER_IS_BETTER:
+    return true;
+  case MetricStrategy::HIGHER_IS_BETTER:
+    return false;
+  default:
+    LBANN_ERROR("Invalid metric strategy!");
+  }
+  return true; // Silence compiler warning about no return.
+}
+
+// Pack model to ship off
+std::string pack(model const& m)  
+{
+  std::ostringstream oss;
+  {
+    RootedBinaryOutputArchive ar(oss, m.get_comm()->get_trainer_grid());
+    ar(m);
+  }
+  return oss.str();
+}
+
+// Send a string to the root of the destination trainer
+void send_string(lbann_comm const& comm, 
+                 std::string const& str, 
+                 int destination_trainer) 
+{
+  size_t size = str.length();
+  comm.send(&size, 1, destination_trainer, /*rank=*/0);
+  comm.send(reinterpret_cast<El::byte const*>(str.data()),
+            size, destination_trainer, /*rank=*/0);
+}
+// Receive a string from the root of src_trainer
+std::string recv_string(lbann_comm const& comm, 
+                        int src_trainer) 
+{
+  size_t size = 0;
+  comm.recv(&size, 1, src_trainer);
+  std::string buf;
+  buf.resize(size);
+  comm.recv(reinterpret_cast<El::byte*>(buf.data()), size, src_trainer);
+  return buf;
+}
+// Unpack received model
+void unpack(model& m, std::string const& str) 
+{
+  std::istringstream iss(str);
+  {
+    RootedBinaryInputArchive ar(iss, m.get_comm()->get_trainer_grid());
+    ar(m);
+  }
+}
+} // namespace
+
+// TruncationSelectionExchange implementation
+
+TruncationSelectionExchange::TruncationSelectionExchange(
+  std::unordered_map<std::string, metric_strategy> metrics, int truncation_k)
+  : m_metrics{std::move(metrics)},m_truncation_k{std::move(truncation_k)}
+{
+  LBANN_ASSERT(m_metrics.size()==1); //only single (1) metric is supported at this time
+}
+
+TruncationSelectionExchange::TruncationSelectionExchange(
+  std::string metric_name,
+  metric_strategy winner_strategy,int truncation_k)
+  : TruncationSelectionExchange({{metric_name, winner_strategy}},truncation_k)
+{}
+
+TruncationSelectionExchange::TruncationSelectionExchange(
+  TruncationSelectionExchange const& other)
+  : m_metrics{other.m_metrics}, m_truncation_k{other.m_truncation_k}
+{}
+
+EvalType TruncationSelectionExchange::evaluate_model(model& m,
+                                       ExecutionContext& ctxt,
+                                       data_coordinator& dc) const
```

Comment:
```suggestion
EvalType TruncationSelectionExchange::evaluate_model(model& m,
                                                     ExecutionContext& ctxt,
                                                     data_coordinator& dc) const
```
Or just run `clang-format`.